### PR TITLE
use oxide-built cockroach v22.1.22

### PR DESCRIPTION
**Pull request by iliana**
This switches us to using builds from our CI setup at https://github.com/oxidecomputer/cockroach.

We could probably use Yet Another Xtask to update the relevant files/checksums to a given commit but I have not done that yet.

### CODE REVIEW

File: tools/cockroachdb_version [hunk 1]
```diff
@@ -1 +1 @@
-v22.1.9
+v22.1.22-25-g24e7b94071
```
Comment (smklein) [inline tools/cockroachdb_version:1]: out of curiosity, what's this new version string mean?

Comment (iliana) [reply tools/cockroachdb_version:1]: This is a [git-describe](https://git-scm.com/docs/git-describe) string, read as "25 commits after the tag `v22.1.22`, at commit `24e7b94071`".

**Review by smklein — APPROVED** [review]
Awesome work on this, this is such smart thing to do to decrease the risk of future issues. Thank you for picking this up and driving it through.

I'm assuming that the `22.1` prefix means this still works with `CockroachDbClusterVersion` defined in `nexus/types/src/deployment/planning_input.rs`?

**Comment by iliana** [comment]
Yeah, the active cluster version doesn't change here, so we don't need to tweak the Reconfigurator policy or any of its related enums at all for this (or likely ever).

Also there is one risk here, which is that we're not running tests on macOS because it takes too long on the GHA runners. (They do presently pass, eventually.) I don't think this is critical enough to block merging this.

**Comment by iliana** [comment]
Oh I should fix the Nix flake to handle this too...

**Comment by david-crespo** [comment]
As discussed offline, this doesn't work on mac yet because the version string built into the binary is constructed differently on the GH runner we use for the mac build. @iliana is fixing it.

```
thread 'main' panicked at /Users/david/oxide/omicron/test-utils/src/dev/mod.rs:124:9:
failed to start CockroachDB: wrong version of CockroachDB installed. expected 'v22.1.22-25-g24e7b94071', found: 'Ok("24e7b94")
stdout:
24e7b94


stderr:


note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  SETUP FAIL [      1/1] crdb-seed: cargo run -p crdb-seed --profile test

   Canceling due to setup script failure
------------
     Summary [   4.832s] 0/1 tests run: 0 passed, 52 skipped
error: setup script failed
 ~/oxide/omicron  [iliana/cockroach-22.1.22]
$ out/cockroachdb/bin/cockroach version
Build Tag:        24e7b94
Build Time:       2024/10/22 17:43:56
Distribution:     OSS
Platform:         darwin amd64 (x86_64-apple-darwin22.6.0)
Go Version:       go1.17.13
C Compiler:       Apple LLVM 15.0.0 (clang-1500.1.0.2.5)
Build Commit ID:  24e7b94071292de7283f5c81c8901e20a00a439c
Build Type:       release
```

**Comment by iliana** [comment]
The above issue will be fixed in https://github.com/oxidecomputer/cockroach/pull/14.

While updating the Nix flake I discovered something utterly horrifying: Debian (and Ubuntu) ship libedit as `libedit.so.2` while the rest of the world uses `libedit.so.0`. I'm going to fix our build to statically link libedit.

**Review by david-crespo — APPROVED** [review]
It works now.

**Pull request merged**